### PR TITLE
[prometheusremotewriteexporter] Support sending trace id and span id for exemplars

### LINKED ISSUE #8351: [prometheusremotewriteexporter] TraceID and SpanID must be added as exemplar attributes.
**Description**
In the [code converting exemplars for PRW](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/488e70172c662f1c32a9ef4e6245c9e650e57327/pkg/translator/prometheusremotewrite/helper.go#L342), we don't do anything with the attached trace ID or span ID

As described in the [prometheus conversion spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification%2Fmetrics%2Fdatamodel.md#exemplars-2), prometheus exporters must attach `trace_id` and `span_id` to prometheus exemplars

### PR DESCRIPTION
**Pull request by dashpole**
**Description:**
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/8351.  This is aligned with the [specification for OTLP -> prometheus](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification%2Fmetrics%2Fdatamodel.md#exemplars-2).

The specification does not say how to handle collisions between attributes named `trace_id` and `span_id` and the trace id or span id of the exemplar itself.  This PR has attributes take precedence over trace id and span id from the exemplar itself.

**Testing:**

Unit test checks that trace id and span id are added as labels on prometheus exemplars.

### CODE REVIEW

File: pkg/translator/prometheusremotewrite/testutils_test.go [hunk 1]
```diff
@@ -218,14 +220,33 @@ func getTimeSeriesWithSamplesAndExemplars(labels []prompb.Label, samples []promp
...
 	e.SetDoubleVal(value)
 	e.SetTimestamp(pdata.NewTimestampFromTime(time))
 	e.FilteredAttributes().Insert(attributeKey, pdata.NewAttributeValueString(attributeValue))
 
+	if traceID != "" {
+		var traceIDBytes [16]byte
+		traceIDBytesSlice, err := hex.DecodeString(traceID)
+		if err != nil {
+			t.Fatalf("error decoding trace id: %v", err)
+		}
```
Comment (mx-psi) [inline pkg/translator/prometheusremotewrite/testutils_test.go:236]:
```suggestion
		require.NoErrorf(t, err, "error decoding trace id: %v", err)
```

File: pkg/translator/prometheusremotewrite/testutils_test.go [hunk 2]
```diff
@@ -218,14 +220,33 @@ func getTimeSeriesWithSamplesAndExemplars(labels []prompb.Label, samples []promp
...
+		copy(traceIDBytes[:], traceIDBytesSlice)
+		e.SetTraceID(pdata.NewTraceID(traceIDBytes))
+	}
+	if spanID != "" {
+		var spanIDBytes [8]byte
+		spanIDBytesSlice, err := hex.DecodeString(spanID)
+		if err != nil {
+			t.Fatalf("error decoding span id: %v", err)
+		}
```
Comment (mx-psi) [inline pkg/translator/prometheusremotewrite/testutils_test.go:245]:
```suggestion
		require.NoErrorf(t, err, "error decoding span id: %v", err)
```

**Review by mx-psi — COMMENTED** [review]
Just one nit to make this more consistent with the rest of the codebase

**Comment by dashpole** [comment]
cc @anneelisabethlelievre, who initially implemented exemplars for PRW
cc @Aneurysm9

**Review by Aneurysm9 — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by dashpole** [comment]
It is non-trivial to rebase (since we now need to handle max exemplar size including these new values), so i'd like another review from someone before we merge.

**Comment by dashpole** [comment]
The additional change is that we have to keep track of the number of runes for the new trace id and span id to make sure the total doesn't exceed limits.  If it does exceed the limit, we need to keep trace and span id, and ignore filtered attributes.  The relevant line from the spec is: `filtered_attributes MUST be added as labels on the OpenMetrics exemplar unless they would exceed the OpenMetrics limit on characters`.  I've added a unit test which tests that case

**Review by Aneurysm9 — APPROVED** [review]

**Comment by dashpole** [comment]
Good to go now!

**Pull request merged**